packages/driver/src/cy/commands/actions/scroll.ts

```diff
@@ -345,23 +324,53 @@ export default (Commands, Cypress, cy, state) => {
           },
         }
 
-        if (!isWin) {
-          log.$el = options.$el
-        }
-
         options._log = Cypress.log(log)
       }
 
-      const ensureScrollability = () => {
-        // Some elements are not scrollable, user may opt out of error checking
-        // https://github.com/cypress-io/cypress/issues/1924
-        if (!options.ensureScrollable) {
-          return
-        }
+      const subjectChain = cy.subjectChain()
 
+      const ensureScrollability = () => {
         try {
-          // make sure our container can even be scrolled
-          return cy.ensureScrollability($container, 'scrollTo')
+          subject = cy.getSubjectFromChain(subjectChain)
```

Comment:
Moved a bunch of these checks asserting on the state of the subject inside the retry loop, so that .scrollTo() can wait for them to pass if they initially fail.